src/sources/http.rs

```diff
@@ -594,6 +668,96 @@ mod tests {
             assert_eq!(log[log_schema().message_key()], "test body".into());
             assert!(log.get(log_schema().timestamp_key()).is_some());
             assert_eq!(log[log_schema().source_type_key()], "http".into());
+            assert_eq!(log["http_path"], "/".into());
+        }
+    }
+
+    #[tokio::test]
+    async fn http_path() {
+        trace_init();
+        let (rx, addr) = source(
+            Encoding::Ndjson,
+            vec![],
+            vec![],
+            "vector_http_path",
+            "/event/path",
+            true,
+        )
+        .await;
+
+        assert_eq!(
+            200,
+            send_with_path(addr, "{\"key1\":\"value1\"}", "/event/path").await
+        );
+
+        let mut events = collect_n(rx, 1).await;
+        {
+            let event = events.remove(0);
+            let log = event.as_log();
+            assert_eq!(log["key1"], "value1".into());
+            assert_eq!(log["vector_http_path"], "/event/path".into());
+            assert!(log.get(log_schema().timestamp_key()).is_some());
+            assert_eq!(log[log_schema().source_type_key()], "http".into());
         }
     }
+
+    #[tokio::test]
+    async fn http_path_no_restriction() {
+        trace_init();
+        let (rx, addr) = source(
+            Encoding::Ndjson,
+            vec![],
+            vec![],
+            "vector_http_path",
+            "/event",
+            false,
+        )
+        .await;
+
+        assert_eq!(
+            200,
+            send_with_path(addr, "{\"key1\":\"value1\"}", "/event/path1").await
+        );
+        assert_eq!(
+            200,
+            send_with_path(addr, "{\"key2\":\"value2\"}", "/event/path2").await
+        );
+
+        let mut events = collect_n(rx, 2).await;
+        {
+            let event = events.remove(0);
+            let log = event.as_log();
+            assert_eq!(log["key1"], "value1".into());
+            assert_eq!(log["vector_http_path"], "/event/path1".into());
+            assert!(log.get(log_schema().timestamp_key()).is_some());
+            assert_eq!(log[log_schema().source_type_key()], "http".into());
+        }
+        {
+            let event = events.remove(0);
+            let log = event.as_log();
+            assert_eq!(log["key2"], "value2".into());
+            assert_eq!(log["vector_http_path"], "/event/path2".into());
+            assert!(log.get(log_schema().timestamp_key()).is_some());
+            assert_eq!(log[log_schema().source_type_key()], "http".into());
+        }
+    }
+
+    #[tokio::test]
+    async fn http_wrong_path() {
+        trace_init();
+        let (_rx, addr) = source(
+            Encoding::Ndjson,
+            vec![],
+            vec![],
+            "vector_http_path",
+            "/",
+            true,
+        )
+        .await;
+
+        assert_eq!(
+            405,
```

Comment:
I realized that 405, "Method not allowed", is a bit of an odd error for this condition. I would have expected a 404.

This looks like a bug in warp: https://github.com/seanmonstar/warp/issues/77

I think you could work around this by, instead of relying on warp's error handling with `path::end`, just accepting all requests and then, in the handler itself, when `strict_match` is true, check that the path matches and return a 404 manually if not.